fs/rc/rcserver/rcserver_test.go

```diff
@@ -847,3 +848,76 @@ func TestServeModTime(t *testing.T) {
 	}}
 	testServer(t, tests, &opt)
 }
+
+func TestContentTypeJSON(t *testing.T) {
+	tests := []testRun{
+		{
+			Name:        "Check Content-Type for JSON response",
+			URL:         "rc/noop",
+			Method:      "POST",
+			Body:        `{}`,
+			ContentType: "application/json",
+			Status:      http.StatusOK,
+			Expected:    "{}\n",
+			Headers: map[string]string{
+				"Content-Type": "application/json",
+			},
+		},
+	}
+	opt := newTestOpt()
+	testServer(t, tests, &opt)
+}
+
+func TestErrorContentTypeJSON(t *testing.T) {
```

Comment:
Couldn't this test just be another test case in ` TestContentTypeJSON`? I'm probably missing something as to why not...

If not I suggest you add a new feature to the testServer rather than duplicating it below. By that I mean add a new flag to `testRun`, set it in the test case, and use it to control the changes you need in `testServer`. That will make for much neater and more maintainable code.